src/Symfony/Component/Security/Core/Authentication/Token/AbstractToken.php

```diff
@@ -26,32 +26,43 @@ abstract class AbstractToken implements TokenInterface
 {
     private $user;
     private $roles = [];
+    private $roleNames = [];
     private $authenticated = false;
     private $attributes = [];
 
     /**
-     * @param (Role|string)[] $roles An array of roles
+     * @param string[] $roles An array of roles
      *
      * @throws \InvalidArgumentException
      */
     public function __construct(array $roles = [])
     {
         foreach ($roles as $role) {
             if (\is_string($role)) {
-                $role = new Role($role);
+                $role = new Role($role, false);
             } elseif (!$role instanceof Role) {
```

Comment:
If we've changed the `$roles` param to be `string[]` instead of `(Role|string)[]` ... shouldn't we trigger some deprecation if this `elseif()` condition is `true`?

Reply:
Since the `Role` class itself is deprecated too the user will have already gotten one deprecation when using it. So as soon as you are free of that deprecation you will pass strings here anyway.